.claude/commands/generate-eng-docs.md

```diff
@@ -0,0 +1,273 @@
+
+  # {{Name}} {{TYPE_CAPITALIZED}}
+
+  Brief introduction paragraph about the {{TYPE}}'s purpose and architecture.
+
+  ## {{TYPE_CAPITALIZED}} Anatomy
+
+  [This section is OPTIONAL - only include for complex {{TYPE}}s]
+  Description of the {{TYPE}}'s structure. For compound {{TYPE}}s, show the hierarchy:
+
+  ```
+  Component.Root
+  ├── Component.Trigger
+  └── Component.Content
+      ├── Component.Item
+      └── Component.Separator
+  ```
+
+  ## {{TYPE_CAPITALIZED}} Usage
+
+  ### How to Import
+
+  ```tsx
+  import { {{Name}} } from '@commercetools/nimbus';
+  ```
+
+  For TypeScript projects, import types:
+  ```tsx
+  import { {{Name}}, type {{Name}}Props } from '@commercetools/nimbus';
+  ```
+
+  ### Basic Example
+
+  ```jsx-live-dev
+  const App = () => (
+    <{{Name}}>
+      Basic implementation example
+    </{{Name}}>
+  )
+  ```
+
+  ### Common props and Variations
+
+  Brief introduction to the most commonly used props.
+
+  #### [Variation 1 - e.g., Size Options, refer to the component source code for the supported variations, if any]
+
+  ```jsx-live-dev
+  const App = () => (
+    <Stack direction="row" gap="400">
+      <{{Name}} size="sm">Small</{{Name}}>
+      <{{Name}} size="md">Medium</{{Name}}>
+      <{{Name}} size="lg">Large</{{Name}}>
+    </Stack>
+  )
+  ```
+
+  #### [Variation 2 - e.g., Variant Options, refer to the component source code for the supported variations, if any]
+
+  ```jsx-live-dev
+  const App = () => (
+    <Stack direction="row" gap="400">
+      <{{Name}} variant="solid">Solid</{{Name}}>
+      <{{Name}} variant="outline">Outline</{{Name}}>
+      <{{Name}} variant="ghost">Ghost</{{Name}}>
+    </Stack>
+  )
+  ```
+
+  [Add more variations as needed based on the {{TYPE}}]
+
+  ## Testing Strategies and Best Practices
+
+  Brief introduction to testing approaches.
+
+  #### 1. Unit Testing with React Testing Library
+
+  ```tsx
+  import { render, screen } from '@testing-library/react';
+  import { {{Name}} } from '@commercetools/nimbus';
+
+  describe('{{Name}}', () => {
+    it('renders correctly', () => {
+      render(<{{Name}}>Test content</{{Name}}>);
+      expect(screen.getByText('Test content')).toBeInTheDocument();
+    });
+  });
+  ```
+
+  #### 2. Storybook Integration for Testing
+
+  Leverage existing stories for visual testing:
+
+  ```tsx
+  export const InteractionTest: Story = {
+    play: async ({ canvasElement }) => {
+      const canvas = within(canvasElement);
+      // Test interactions based on {{TYPE}} behavior
+    },
+  };
+  ```
+
+  #### 3. Regression tests
+
+  **[Placeholder for regression test strategy]**
+  - Visual regression with Chromatic
+  - Behavior regression with interaction tests
+  - API contract testing
+
+  ### 4. Accessibility requirements
+
+  - Keyboard navigation: [Describe keyboard patterns]
+  - Screen reader support: [Describe announcements]
+  - ARIA attributes: [List key ARIA attributes]
+  - Color contrast: Meets WCAG AA standards
```

Comment:
These a11y requirements are already handled in the design guidelines. This section should instead be a part of the usage section, and document what configuration is necessary for consumers to implement to ensure accessibility